### ISSUE TITLE
Databricks Infrastructure Encryption cannot be used in conjunction with Customer Managed Keys

### ISSUE THREAD

**Issue by istairbn** [terraform-providers/terraform-provider-azurerm#12615]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

By design, you should be able to enable customer managed keys *and* Infrastructure Encryption. Current implementation denys the attempt to enable both. 

}

### Potential Terraform Configuration

```hcl

resource "azurerm_databricks_workspace" "example" {
  name                = "databricks-test11"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  sku                 = "premium"
  customer_managed_key_enabled = true
  infrastructure_encryption_enabled = true

}

```

### References

* #0000

**Comment by istairbn** [comment]
#12614 should fix this

**Comment by tombuildsstuff** [comment]
Fixed via #12614
